### ISSUE TITLE
🐛 Bug Report: Failed to parse page JSON data: expected value at line 1 column 1 (11/18)

### ISSUE THREAD

**Issue by sigaloid** [redlib-org/redlib#324]
## This issue has been fixed and a new image has been released. Pull and update - do not comment here saying it's not fixed unless your error page says "✅ Instance is up to date."

**Comment by sigaloid** [comment]
@grngxd You mentioned some instances are up and working, which ones?

**Comment by tiphae** [comment]
My Redlib is giving this error, but so is my backup Libreddit that usually works when Redlib doesn't. I am able to view Reddit directly just fine (from same IP).

**Comment by grngxd** [comment]
> @grngxd You mentioned some instances are up and working, which ones?

https://libreddit.privacydev.net/ was working when i tested it, its dead now.

**Comment by grngxd** [comment]
@sigaloid

**Comment by sigaloid** [comment]
Ah, possibly still being cached.

**Comment by Coccocoahelper** [comment]
From my 1-minute testing, no instance seems to be functional. Still going to check more, though.

**Comment by gigirassy** [comment]
Thought I was IP banned, but looks like this applies to all Redlib instances.
Here's my WebDock KVM VPS' Redlib instance (basic auth), displaying this error:
![image](https://github.com/user-attachments/assets/2c3be4e5-eeb1-4bd1-bc09-2d7ab7941dd0)
And here's a test container on my regular main Windows PC:
![image](https://github.com/user-attachments/assets/638ad125-c8b3-4d8b-b8b6-7111d765b43e)
VPS logs:
![image](https://github.com/user-attachments/assets/21a462e5-e364-40bb-8873-9cafc2b1cfd5)
Windows PC Docker Desktop logs:
![image](https://github.com/user-attachments/assets/500ff080-f10f-452e-8d3a-a86b1e686f2d)

**Comment by Coccocoahelper** [comment]
> Thought I was IP banned, but looks like this applies to all Redlib instances. Here's my WebDock KVM VPS' Redlib instance (basic auth), displaying this error: ![image](https://private-user-images.githubusercontent.com/145225213/387391752-2c3be4e5-eeb1-4bd1-bc09-2d7ab7941dd0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5NjQ5MjMsIm5iZiI6MTczMTk2NDYyMywicGF0aCI6Ii8xNDUyMjUyMTMvMzg3MzkxNzUyLTJjM2JlNGU1LWVlYjEtNGJkMS1iYzA5LTJkN2FiNzk0MWRkMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOFQyMTE3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MDBjNzAyMzVmYzczNzg4N2YzOGY2ODA5ZjI2NjJiOTYyMzc3ZmEwYmM2N2YzOTE0ZmM2MmZiNmFjZTcxN2M1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.38dw-E_UgL4mm7aHk-4pEVF8mYj5xwwyjs8l2TTioqo) And here's a test container on my regular main Windows PC: ![image](https://private-user-images.githubusercontent.com/145225213/387392058-638ad125-c8b3-4d8b-b8b6-7111d765b43e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5NjQ5MjMsIm5iZiI6MTczMTk2NDYyMywicGF0aCI6Ii8xNDUyMjUyMTMvMzg3MzkyMDU4LTYzOGFkMTI1LWM4YjMtNGQ4Yi1iOGI2LTcxMTFkNzY1YjQzZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOFQyMTE3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZGUzZGUxOGZkMmRhZGQxNTM4NmE1M2ZmNDgxYzcyNTBiZTYxNDUxZTJkMGZlZTdjNDIyNmQ0MDlmMjY2MGQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.tu5aq6RgEODBA0tUEVhkNWN2e0EehMqZ73PF1RTEhYc) VPS logs: ![image](https://private-user-images.githubusercontent.com/145225213/387392325-21a462e5-e364-40bb-8873-9cafc2b1cfd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5NjQ5MjMsIm5iZiI6MTczMTk2NDYyMywicGF0aCI6Ii8xNDUyMjUyMTMvMzg3MzkyMzI1LTIxYTQ2MmU1LWUzNjQtNDBiYi04ODczLTljYWZjMmIxY2ZkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOFQyMTE3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNjRlN2Q3M2ZiNWNkMzdjOWViMzE4ZjVkZjA1Y2UyMGQxNjcwMDhlZDM1MWEwNjQ2NWZkNGZhYjA0NjcxOGJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GsqM-qpQOVng3LfCY3eJyyCUpqEsABjSsoYDnbvNeUQ) Windows PC Docker Desktop logs: ![image](https://private-user-images.githubusercontent.com/145225213/387392237-500ff080-f10f-452e-8d3a-a86b1e686f2d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5NjQ5MjMsIm5iZiI6MTczMTk2NDYyMywicGF0aCI6Ii8xNDUyMjUyMTMvMzg3MzkyMjM3LTUwMGZmMDgwLWYxMGYtNDUyZS04ZDNhLWE4NmIxZTY4NmYyZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOFQyMTE3MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OTc3OGU2ZWU3NjY5MjZiM2Y4ZWU3YjY1MmVjYjM1YzM3YzAxNTk1ZjBkM2RmMDNhMDcyZGZhZTA4ZGVhYjZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WKX22ad_ctnSGgsMsEcdFWp9XfPBAoclF0n6aYYtBDM)

Same thing here. But another frontend, [Eddrit](https://edrit.com) ([repo](https://github.com/corenting/eddrit)) does work, for some reason.

**Comment by argium** [comment]
Is the process for debugging this documented?  I'd like to dump the underlying request/response.

**Comment by e455a81e-d3ba-41a2-bc6d-7aafb1d9a5cd** [comment]
I just noticed this with my private instance. Interestingly enough, queries to the json endpoints succeed with curl but fail in redlib.

**Comment by sigaloid** [comment]
Unfortunately there's not much in terms of debugging. There was a PR up a few weeks back to disable TLS verification which would be the first step but we would also need to add proxy support. We're on an old version of hyper so it might be a pain to do that. 

It's been on my back burner for a while because it's trivial to issue the two exact requests Redlib sends, and if it works, it's definitely a TLS signature error.

good news is that I'm fairly certain that the issue is another TLS fingerprinting issue because issuing the same requests independent of our TLS stack works just fine. So I might be forced to upgrade that anyway in the course of fixing it. Bad news is, of course, that this means it's officially a cat and mouse game between me and some Reddit SWE. That just hurts - other frontends like previously mentioned aren't being targeted like Redlib is! I guess it's because we're the most popular. :( 

Anyway. Working on finding a fix now

**Comment by maxexcloo** [comment]
This has to be one of the best supported selfhosted apps I use - I always wake up, notice Reddit isn't working, have a coffee and notice it's fixed!

Always appreciate the hard work @sigaloid ☕

**Comment by Bumpf** [comment]
https://reddit.privacy.com.de/ is throwing this error on the VPN IP I am usually on, and also on my home IP! I realized you said to check a second IP after I had already filled out the info but maybe this is useful anyway (it's for the VPN).

1. Confirmed
2. All requests seem to return `Failed to parse yadda yadda` [I checked the main page, a subreddit, and a specific post. 3 URLs means I can't access any page, right? ;) ]
3. IPv4
4. AS No. 396253, IP:149.19.41.176

**Comment by argium** [comment]
@sigaloid I added some instrumentation and ran it locally. 

![image](https://github.com/user-attachments/assets/1c00c8a6-6f59-40ef-8108-ae339c3eb77a)

Are you working by trial and error?  Do you run the real reddit app and then compare?

PS. Have we considered filing a support ticket? 😂

**Comment by sigaloid** [comment]
I test with the two basic requests that tell me where in the stack in the error is; basically just token request then /r/popular with the token auth (matching the same headers and all). If that works in python/curl, I know the issue is within the TLS stack.

That network policy error is pretty common, back when the API was easier to work with it happened when you had certain messages in your user agent. Nowadays it means something in your request is making the edge CDN's and WAF's concerned.

**Comment by argium** [comment]
> I test with the two basic requests that tell me where in the stack in the error is; basically just token request then /r/popular with the token auth (matching the same headers and all). If that works in python/curl, I know the issue is within the TLS stack.

Is that what's happening today too? 

FWIW, they could just be targeting third party use generally. I'm sure they have a lot of bots to contend with that cause them significantly worse headaches. 

They may have also pushed out a new app apk a couple days ago and I wonder whether they're making corresponding changes there too.

**Comment by lvxnull2** [comment]
Even the pre-#301 commits of redlib are broken, they worked previously after frequent restarts

**Comment by sigaloid** [comment]
> Shows Error 500, so it's probably also blocked.

Ah, phew, glad it's not just redlib :grin: 

> Is that what's happening today too?

Yes. I can recreate the requests without failures on another TLS stack in Rust as well. So I know it's a `hyper_rustls` issue.

> They may have also pushed out a new app apk a couple days ago and I wonder whether they're making corresponding changes there too.

I don't know if they'd break new apps that quickly. I hope not, as that'll really increase my maintenance workload. 

Anyway, good news all, I've confirmed a fix in a96bebb0991558ff278b4cabf3c54678d7d8b208 by switching to `hyper-tls` which uses `native-tls` which uses a vendored `openssl-dev`. A few downsides: not a rustic TLS stack, so it makes cross-compiling and static-compiling a bit harder and less predictable. Also it's slightly larger binary-wise. But I can confirm the latest docker image at `quay.io/redlib/redlib:latest` **does** work :)

**Comment by donslice** [comment]
Just pulled and restarted. Can confirm working on my self-hosted instance. Thanks for being so responsive!

**Comment by geraldwuhoo** [comment]
Thanks for your hard work on this. Unfortunately, it looks like the cross-compilation is broken: https://github.com/redlib-org/redlib/actions/runs/11917620723/job/33213127030. Realized when I pulled the latest on my ARM server and it was still out-of-date. Any chance for a fix?

**Comment by sigaloid** [comment]
Yep, arm build is unfortunately broken due to the switch. I'll be looking at this when I get a chance.

**Comment by 3ster** [comment]
I am still getting this error on the latest version (sha-96e40e888742):
```
Failed to parse page JSON data: expected value at line 1 column 1 | /r/popular/hot.json?&raw_json=1&geo_filter=GLOBAL
[Reddit Status](https://www.redditstatus.com/)

✅ Instance is up to date.

Expected something to work? [Report an issue](https://github.com/redlib-org/redlib/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=%F0%9F%90%9B+Bug+Report%3A+Failed%20to%20parse%20page%20JSON%20data:%20expected%20value%20at%20line%201%20column%201%20|%20/r/popular/hot.json?&raw_json=1&geo_filter=GLOBAL)

If you're getting a "Failed to parse page JSON data" error, please check [#318](https://github.com/redlib-org/redlib/issues/318)

Head back [home]()?
```
Instance is running on a residential IP and CURLing reddit / accessing reddit in browser works fine.

**Comment by sigaloid** [comment]
Can you by chance test on another IP? This sounds like #318

**Comment by Kakifrucht** [comment]
same issue, tried my local ip, vpn ip and datacenter ip. same error on all, latest commit build

**Comment by 3ster** [comment]
Grabbed a new v4 (and v6 prefix), same issue - don't think this is IP-related, unless reddit decided to ban my entire ISP.

**Comment by sigaloid** [comment]
Yep, it's back. Okay, picking it back up again

**Comment by donslice** [comment]
Unfortunately I can confirm I'm getting the issue again as well. As of about 5 minutes ago. 

Running Redlib v0.35.1 on [::]:8080!
 ERROR redlib::client > Got an invalid response from reddit expected value at line 1 column 1. Status code: 403 Forbidden
 ERROR redlib::utils  > Error page rendered: Failed to parse page JSON data: expected value at line 1 column 1 
 ERROR redlib::client > Got an invalid response from reddit expected value at line 1 column 1. Status code: 403 Forbidden
 ERROR redlib::utils  > Error page rendered: Failed to parse page JSON data: expected value at line 1 column 1 
Starting Redlib...
 INFO  redlib > Evaluating config.
 INFO  redlib > Evaluating instance info.
 INFO  redlib > Creating OAUTH client.
 INFO  redlib::oauth > [🔄] Spoofing Android client with headers: {"X-Reddit-Device-Id": "727586a9-41e2-4bb2-94de-2d2c3812de99", "Client-Vendor-Id": "727586a9-41e2-4bb2-94de-2d2c3812de99", "User-Agent": "Reddit/Version 2023.11.0/Build 830610/Android 13"}, uuid: "727586a9-41e2-4bb2-94de-2d2c3812de99", and OAuth ID "ohXpoqrZYub1kg"
 TRACE redlib::oauth > Sending token request...
 TRACE redlib::oauth > Received response with status 200 OK and length Some("1308")
 TRACE redlib::oauth > Serializing response...
 TRACE redlib::oauth > Accessing relevant fields...
 INFO  redlib::oauth > [✅] Success - Retrieved token "eyJhbGciOiJSUzI1NiIsImtpZCI6IlNI...", expires in 86399
 INFO  redlib::oauth > [✅] Successfully created OAuth client
 INFO  redlib::oauth > [⏳] Waiting for 86279s seconds before refreshing OAuth token...
Running Redlib v0.35.1 on [::]:8080!
 ERROR redlib::client > Got an invalid response from reddit expected value at line 1 column 1. Status code: 403 Forbidden
 ERROR redlib::utils  > Error page rendered: Failed to parse page JSON data: expected value at line 1 column 1

**Comment by sigaloid** [comment]
okay... really weird behavior. I can't tell if I'm being trolled because I am **100%** certain my fix worked. And the only thing that I did just now to fix it (3e1718bfc971017aacee180d497f398be2d13855) was remove some header that really should not be causing any issues. Seriously the official app sends Accept-Language. I have to assume this is manual actions being taken by Reddit, plucking out individual requests we're sending and trying to fingerprint them.

Anyway, fix is out. I can confirm with certainty it works at this moment across the world. Whether it gets blocked in the next 10 minutes, no one can really say :thinking:

**Comment by edgar-vincent** [comment]
I'm sorry to say that I'm being affected by the issue again with the latest commit (instance is said to be up to date).
Thanks a lot for your work.

**Comment by manuquadrat** [comment]
> I'm sorry to say that I'm being affected by the issue again with the latest commit (instance is said to be up to date). Thanks a lot for your work.

My local instance is running without any issues. So it could be you are facing an IP block?

**Comment by sigaloid** [comment]
Yeah, I cannot reproduce anywhere, it's either an IP ban (#318) or we'll see in ten minutes whether or not it'll be another incident

**Comment by unclearParadigm** [comment]
And we're blocked again. :latest, x86, not ip-banned.

**Comment by BigBoyBarney** [comment]
It worked for a few minutes for me as well, and it's blocked again. Interestingly, I'm getting an "instance not up to date" error as well, even though I'm 100% it's from latest `main`.
![image](https://github.com/user-attachments/assets/913328ef-5d21-404e-81da-6781c8f8a9e5)

**Comment by wuchyi** [comment]
Same on my end as well. Running the latest docker container (sha-0bc36d5)

**Comment by sigaloid** [comment]
Yeppp. They're for sure trying to block us now. Everyone wave to the employee monitoring our commits and issues.

On it again

**Comment by revilo464** [comment]
Well, I have nothing relevant to add... But this is an interesting cat and mouse game.

I will keep killing, removing, and pulling the latest image accordingly.

*grabs popcorn*

**Comment by sigaloid** [comment]
Aw man, this is so funny. Okay, new commit out (18efb8c714ad10e0082059e1d47e0f95686d04a7). Image coming soon. Please silence your phones and enjoy the show.

**Comment by sooox-cc** [comment]
I have a one person instance so I dont think its a ban tbh

`⚠️ This instance is not up to date and is 12 commits old. Test and confirm on an up-to-date instance before reporting.`

**Comment by Kakifrucht** [comment]
as long as they are not just outright banning ip's specifically or locking down their anonymous api's, would it not be possible to reach perfect parity with the "legit" allowed packets that reddit uses for their frontends when not logged in?

**Comment by mattfbacon** [comment]
> So I'm pretty sure they're updating from the list.

So generate random extra headers to include with each request? Not long-term but until we can resolve #327 there's not much we can do

**Comment by sigaloid** [comment]
That is what I did in my last commit, but I'm also shuffling the ordering (which it turns out is the real fingerprinting that they're doing on us). 6ecdedd

**Comment by 0x62c734ec** [comment]
https://github.com/curl/curl/issues/9362 check this.

**Comment by Patriot2407** [comment]
Interestingly, I'm getting an "instance not up to date" error as well, even though I'm 100% it's from latest `main`.

I'm just a normie and new around here, but my non-educated guess is that this is because the docker image may be a few commits behind the latest code commit. The docker image is probably not being rebuilt and published every time there is a new commit (reasonably so). Feel free to correct me if I'm off my rocker.

**Comment by NTchrist** [comment]
Showerthought:

what if we changed the license to something that prohibited companies with more than $100m in revenue from reading any code from the repository without paying $10m for a license, change the fingerprinting again, know that they are violating the license by reading your commits, sue them for said $10m dollars, and during discovery get them to divulge how their fingerprinting works?

( I don't have anything actually constructive to add, so hopefully I at least made someone laugh.  We're rooting for you @sigaloid ! )

**Comment by sigaloid** [comment]
> The docker image is probably not being rebuilt and published every time there is a new commit (reasonably so). Feel free to correct me if I'm off my rocker.

It's actually built on every single commit, so if you run into any errors, the first thing you should do is pull a new version (that's why I added that message). Currently the latest version does work, so definitely pull it.

**Comment by binamov** [comment]
Y'all [get this man some caffeine ☕](buymeacoffee.com/sigaloid)

**Comment by FireMasterK** [comment]
> good news is that I'm fairly certain that the issue is another TLS fingerprinting issue because issuing the same requests independent of our TLS stack works just fine. So I might be forced to upgrade that anyway in the course of fixing it. Bad news is, of course, that this means it's officially a cat and mouse game between me and some Reddit SWE. That just hurts - other frontends like previously mentioned aren't being targeted like Redlib is! I guess it's because we're the most popular. :(

I think it's most likely a TLS fingerprinting issue too, we should replicate the TLS fingerprint from the Android client. We should copy the TLS extensions using something like Wireshark, and try to replicate it. :)

**Comment by WreckingBANG** [comment]
> Y'all [get this man some caffeine ☕](https://buymeacoffee.com/sigaloid)

Do it! I did my part! o7

**Comment by alangrafu** [comment]
Thanks for all your effort!

**Comment by maxexcloo** [comment]
Appreciate the work again! Some sleuthing indeed 🕵🏻

**Comment by sigaloid** [comment]
Thanks for all of the kind words and donations. Very motivational during these times of high pressure :)

**Comment by maxexcloo** [comment]
Issue seems to be back again on my instance unfortunately...

**Comment by sigaloid** [comment]
Unable to reproduce, perhaps give it a few minutes?

**Comment by maxexcloo** [comment]
> Unable to reproduce, perhaps give it a few minutes?

Still unavailable for me, it was working fine till I posted the message - is there any way I can debug?

**Comment by sigaloid** [comment]
Try from another IP - I'm not getting it residentially nor commercially. It may be a widespread issue just taking longer to reach the edge CDN servers I'm nearest. But more than likely #318

**Comment by sooox-cc** [comment]
its working fine for me after pulling the latest docker image

**Comment by maxexcloo** [comment]
> Try from another IP - I'm not getting it residentially nor commercially. It may be a widespread issue just taking longer to reach the edge CDN servers I'm nearest. But more than likely #318

Yup it was an IP ban, thanks for the tip - have posted a workaround with Cloudflare WARP inside docker compose here: https://github.com/redlib-org/redlib/issues/318#issuecomment-2490882093

**Comment by dave0003** [comment]
For what it's worth, I tried the latest build:  6be6f89 11/21/2024
And immediately experienced this problem again.  I reverted back to:  d3ba5f3 11/19/2024
And this slightly older version is still working.

I'm running the binary:  redlib-x86_64-unknown-linux-musl

**Comment by sigaloid** [comment]
That's really weird, nothing in those commits should have changed anything (https://github.com/redlib-org/redlib/compare/d3ba5f3efb6825f4b0454523dae472b2adda3066...6be6f892a4eb159f5a27ce48f0ce615298071eac).

Removal of X-Reddit-Device-Id perhaps, but that no longer is sent from my tests...

 I can't reproduce. I'd say it's #318 but if it's working on an older version...

Is this reproducible consistently? There's some things that are set non-deterministically in the spoofer.

**Comment by dave0003** [comment]
It's consistent.  I just downloaded the latest build again, and immediately encountered the error.  Reverted to the older version again and it's fine.

**Comment by sigaloid** [comment]
I wouldn't recommend downloading builds, the release system is a little weird and the releases get overwritten by new commit builds (but not the one labeled `redlib` in the release). Been meaning to work on that for a while.

I would recommend cloning the code and checking out the commits in question (right now would be good to test `main`), with `RUST_LOG=redlib=trace cargo run`. That, or using the container images over at https://quay.io/redlib/redlib

**Comment by Handrail9** [comment]
I'm having this issue on the latest pull. I run the docker container and it looks like my residential IP address might be blocked? I'm being told to sign in if I visit regular Reddit. Let me know if theres any more info I can provide or if Ive done Something wrong

**Comment by AyoungDukie** [comment]
Encountering this, and it seems like a new push of banning again, as many of the public instances are also going down:

https://stats.uptimerobot.com/mpmqAs1G2Q
![Screenshot_20241205-235717](https://github.com/user-attachments/assets/2377e9e1-e850-4aec-988f-9ed56864a828)
![IMG_20241206_000116](https://github.com/user-attachments/assets/f40a2fd1-cc3c-4268-9cc6-55a3d5cdd484)
![IMG_20241206_000136](https://github.com/user-attachments/assets/c3862a3e-a3b5-4cce-a7fd-2c82f92c5650)

**Comment by satonotdead** [comment]
I'm the only one getting `⚠️ Error checking update status.`?

Already updated to last commit and restarted the intance.

**Comment by sigaloid** [comment]
Either means it failed to fetch the instance list from GitHub, or it failed to fetch the commit list from GitHub - sounds like your IP has been limited somehow by GitHub.

**Comment by scriptsandthings** [comment]
Is there a max number of subreddits one can use in redlib?

Along with the failed to parse message, I also receive a "URI Too Long" error once adding additional subreddits. If I remove the subreddits from my docker container and restart it, the error goes away and redlib works again.

Not sure how many subreddits tipped it over the edge, but I was attempting to use a total of 936 subreddits when I encountered the issue. Cutting the number in half to 468 resolved the URI length message.

Assuming this limit would be on reddit's side, not anything redlib could work around without sending multiple requests after x subreddits have been added.

> redlib  |  ERROR redlib::client > Got an invalid response from reddit expected value at line 1 column 1. Status code: 414 URI Too Long
> redlib  |  ERROR redlib::utils  > Error page rendered: Failed to parse page JSON data: expected value at line 1 column 1

**Comment by Tokarak** [comment]
Question: we're shuffling headers, but isn't there a chance that the headers end up in the original position? Isn't that a risk?

**Comment by groetg** [comment]
Whats the solution? Everytime I update my docker container I get this error and have to start allover with a new docker container (it seems)

**Comment by freedit-dev** [comment]
> Whats the solution? Everytime I update my docker container I get this error and have to start allover with a new docker container (it seems)

https://git.ptr.moe/baalajimaestro/redlib

**Comment by groetg** [comment]
Can't get that docker running. Permission error

<img width="1080" height="2424" alt="Image" src="https://github.com/user-attachments/assets/1e9825b5-3c54-46e7-871d-e3bfb0940d15" />
